### PR TITLE
Fixes and tidying

### PR DESCRIPTION
**Pull request by amaanq**
- **PR Description**

I applied a lot of changes suggested by gopls and staticcheck. The main changes are rearranging struct fields to use less space in memory, reorganizing imports to separate local project imports from others, use _ for unused parameters, and applying DeMorgan's law for cleaner expressions. 

I also tidied up the docs and formatted them as per markdownlint suggestions, as well as the ci scripts.

A lot of these changes **are** subjective, but I hope you'll appreciate them :grin: 

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by jesseduffield** [comment]
I'm glad you're liking Lazygit. In the past I've had bad experiences with PRs that make a bunch of changes using an automated tool, so I'm going to be strict here.

Things I don't want:
* changes to non-go files
* rearranging struct fields
* rearranging imports
* forcing the use of an embedded struct (e.g. 'gitCommon.Common.UserConfig = deps.userConfig' -> 'gitCommon.UserConfig = deps.userConfig')
* replacing unused args with underlines

Things I'm okay with:
* Replace -> ReplaceAll
* if statements to switch statements
* having struct fields each on their own line
* de morgan's law (although I didn't see any examples of this)

As for that changes I'm okay with, I only want to make these changes if the rule can be added to golangci (see `.golangci.yml`). If they can't, there's little point in making the changes because inconsistencies will arise in the future.

Likewise I'd be okay if organising imports could be done upon hitting save (either by gofmt or gofumpt) but I haven't looked into whether that's a thing.

**Comment by amaanq** [comment]
The organized imports can be done upon save with goimports-reviser, with that knowledge should I keep the organized imports then? I'm not sure if golangci can catch that

**Comment by jesseduffield** [comment]
Let's keep that out of this PR. We can revisit in another PR

**Comment by jesseduffield** [comment]
@amaanq do you still intend to get back to this PR? You'll need to rebase onto the latest master as there are conflicts

**Comment by amaanq** [comment]
yeah, rebased & removed stuff you didn't want in

File: pkg/commands/git_commands/deps_test.go [hunk 1]
```diff
@@ -45,9 +46,9 @@ func buildGitCommon(deps commonDeps) *GitCommon {
 	}
 	gitCommon.cmd = cmd
 
-	gitCommon.Common.UserConfig = deps.userConfig
-	if gitCommon.Common.UserConfig == nil {
-		gitCommon.Common.UserConfig = config.GetDefaultConfig()
+	gitCommon.UserConfig = deps.userConfig
```
Comment (jesseduffield) [inline pkg/commands/git_commands/deps_test.go:49]: I don't want any changes related to embedded structs such as this

File: pkg/gui/context/list_context_trait.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ type ListContextTrait struct {
 
 	c                 *ContextCommon
 	list              types.IList
-	getDisplayStrings func(startIdx int, length int) [][]string
```
Comment (jesseduffield) [inline pkg/gui/context/list_context_trait.go:15]: I don't want any omitted types

File: pkg/gui/context/working_tree_context.go [hunk 3]
```diff
@@ -23,7 +24,7 @@ func NewWorkingTreeContext(c *ContextCommon) *WorkingTreeContext {
 		c.UserConfig.Gui.ShowFileTree,
 	)
 
-	getDisplayStrings := func(startIdx int, length int) [][]string {
```
Comment (jesseduffield) [inline pkg/gui/context/working_tree_context.go:26]: I don't want this either

File: pkg/gui/presentation/graph/graph.go [hunk 4]
```diff
@@ -348,7 +348,7 @@ func renderPipeSet(
 	}
 
 	for _, pipe := range nonSelectedPipes {
-		if pipe.kind != STARTS && !(pipe.kind == TERMINATES && pipe.fromPos == commitPos && pipe.toPos == commitPos) {
```
Comment (jesseduffield) [inline pkg/gui/presentation/graph/graph.go:351]: I find the existing form easier to read

File: pkg/i18n/dutch.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ func dutchTranslationSet() TranslationSet {
 		OnLatestVersionErr:                  "Je hebt al de laatste versie",
 		MajorVersionErr:                     "Nieuwe versie ({{.newVersion}}) is niet backwards compatibele vergeleken met de huidige versie ({{.currentVersion}})",
 		CouldNotFindBinaryErr:               "Kon geen binary vinden op {{.url}}",
-		IntroPopupMessage:                   "Bedankt voor het gebruik maken van lazygit! 2 dingen die je moet weten:\n\n1) Als je meer van lazygit zijn features wilt leren bekijk dan deze video:\n   https://youtu.be/CPLdltN7wgE\n\n2) Als je git gebruikt, ben je een programmeur! Met jouw hulp kunnen we lazygit verbeteren, dus overweeg om een ​​donateur te worden en mee te doen aan het plezier op\n   https://github.com/jesseduffield/lazygit",
+		IntroPopupMessage:                   "Bedankt voor het gebruik maken van lazygit! 2 dingen die je moet weten:\n\n1) Als je meer van lazygit zijn features wilt leren bekijk dan deze video:\n   https://youtu.be/CPLdltN7wgE\n\n2) Als je git gebruikt, ben je een programmeur! Met jouw hulp kunnen we lazygit verbeteren, dus overweeg om een donateur te worden en mee te doen aan het plezier op\n   https://github.com/jesseduffield/lazygit",
```
Comment (jesseduffield) [inline pkg/i18n/dutch.go:115]: What has changed here?

**Review by jesseduffield — CHANGES_REQUESTED** [review]
See comments

**Comment by jesseduffield** [comment]
Closing

**Pull request closed without merging**